react-validated-input/my-app/src/App.tsx

```diff
@@ -0,0 +1,54 @@
+import { useState } from 'react';
+import { FaCheck } from 'react-icons/fa';
+
+import './App.css';
+import { IconType } from 'react-icons';
+
+function App() {
+  const [password, setPassword] = useState('');
+
+  function handlePassword(valid: string | IconType, notValid: string): string {
+    return password.length === 0
+      ? notValid
+      : password.length < 8
+      ? notValid
+      : /^(?=.*\d)(?=.*[A-Z])(?=.*[!@#$%^&*()]).+$/.test(password)
+      ? valid
+      : notValid;
```

Comment:
A lot of people dislike these nested ternaries like this since they can be hard to read. They find if/else easier to read. I haven't decided which way I prefer, though.